presto-main/src/main/java/com/facebook/presto/type/BigintOperators.java

```diff
@@ -195,8 +195,10 @@ public static double castToDouble(@SqlType(StandardTypes.BIGINT) long value)
         return value;
     }
 
+    // x <= 20
     @ScalarOperator(CAST)
-    @SqlType(StandardTypes.VARCHAR)
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
```

Comment:
I've marked this as FIXME.
There is a need to implement possibility of passing literal long parameters as function parameter as prerequirement for fixing this.